scripts/upload.py

```diff
@@ -32,14 +34,25 @@ def upload(globpath: str, container: str, queue: str, sas_token_env: str, storag
         credential = None
         try:
             dac = DefaultAzureCredential()
-            credential = ClientAssertionCredential(TENANT_ID, CLIENT_ID, lambda: dac.get_token("api://AzureADTokenExchange/.default").token)
+            credential = ClientAssertionCredential(TENANT_ID, ARC_CLIENT_ID, lambda: dac.get_token("api://AzureADTokenExchange/.default").token)
             credential.get_token("https://storage.azure.com/.default")
         except ClientAuthenticationError as ex:
-            getLogger().info("Unable to use managed identity. Falling back to environment variable.")
-            credential = os.getenv(sas_token_env)
+            credential = None
+            getLogger().info("Unable to use managed identity. Falling back to certificate.")
+            cmd_line = list((os.path.join(str(helixpayload()), 'certhelper', 'CertHelper.exe')))
+            cert_helper = RunCommand(cmd_line, None, False, 0)
+            cert_helper.run()
+            for cert in cert_helper.stdout.splitlines():
+                credential = CertificateCredential(TENANT_ID, CERT_CLIENT_ID, certificate_data=base64_to_bytes(cert))
+                try:
+                    credential.get_token("https://storage.azure.com/.default")
+                except ClientAuthenticationError as ex:
+                    credential = None
+                    continue
         if credential is None:
-            getLogger().error("Sas token environment variable {} was not defined.".format(sas_token_env))
-            return 1
+            getLogger().error("Unable to authenticate with managed identity or certificates.")
+            getLogger().info("Falling back to environment variable.")
+            credential = os.getenv(sas_token_env)
```

Comment:
If we are going to keep the envvar credential, I think we should also keep the check to see if it is None that it had before.

Reply:
Although I could see us removing the SAS token functionality entirely once we have everything setup.

Reply:
That was my plan. Is that we would be getting rid of this fallback in a couple of days.